remove specific experiments in queue (#2715)
iterative/dvc.org: content/docs/command-reference/exp/remove.md

## Description

Deletes one or more experiments, indicated by name (see `dvc exp run`) or

> revision (only queued experiments).

Reviewer: ```suggestion
commit SHA (only for queued experiments).
```

> commit SHA (only queued experiments).

Reviewer: > commit SHA (only queued experiments)

So regular experiments can't be removed by commit SHA @karajan1001 ?

Reviewer: Thanks for the update, BTW!

Reviewer: @karajan1001 can regular commits be removed by SHA? Thanks

> revision(only for experiments in the queue).

Reviewer: ```suggestion
revision (only queued experiments).
```

instead (shown when queued and by `dvc exp show`).

Let's queue a few experiments in different ways and then delete some of them:

```dvc
$ dvc exp run --queue -S train.min_split=64
Queued experiment 'e41d5b4' for future execution.
$ dvc exp run --queue -S train.min_split=32 --name split32
Queued experiment '5751540' for future execution.
$ dvc exp run --queue -S train.min_split=16 --name split16
Queued experiment '8de9a6c' for future execution.

$ dvc exp e41d5b4 split16

$ dvc exp show --include-params=train.min_split --no-pager
```

```table
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Experiment            ┃ Created      ┃ State  ┃ avg_prec ┃ roc_auc ┃ train.min_split ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ workspace             │ -            │ -      │  0.57553 │ 0.94652 │ 2               │
│ master                │ Aug 02, 2021 │ -      │  0.53252 │  0.9107 │ 2               │
│ └── 5751540 [split32] │ 04:57 PM     │ Queued │        - │       - │ 32              │
└───────────────────────┴──────────────┴────────┴──────────┴─────────┴─────────────────┘

> ```

Reviewer: I think we should use `exp list` instead to simplify this example. `exp show` can be noisy.

Author: I tried on my computer that `dvc list` could not show queued exps.

Reviewer 2: Indeed, neither `dvc exp list` nor `dvc exp list --all` shows the queued experiments. It looks the only way to get the list of queued experiments is via `dvc exp show`.

Reviewer: OK let's keep the `exp show` table then. Thanks

BTW should `exp list` print queued exps? Is there an issue in the core repo perhaps?

Reviewer 3: TBH I think the best solution is moving all of this queue functionality into a `queue` command as suggested in https://github.com/iterative/dvc/issues/5615. I have avoided it because it's a bigger change, but I think it will be needed soon, especially now that we are working on remote execution.

```dvc
$ dvc exp remove --queue
$ dvc exp show --include-params=train.min_split --no-pager
┏━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Experiment ┃ Created      ┃ avg_prec ┃ roc_auc ┃ train.min_split ┃
┡━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ workspace  │ -            │  0.57553 │ 0.94652 │ 2               │
│ master     │ Aug 02, 2021 │  0.53252 │  0.9107 │ 2               │
└────────────┴──────────────┴──────────┴─────────┴─────────────────┘

> ```

Reviewer: Let's delete this and create an **Example: remove all experiments** in https://github.com/iterative/dvc.org/pull/2722 instead, and throw this in there?

$ dvc exp run --queue -S train.min_split=16 --name split16
Queued experiment '8de9a6c' for future execution.
$ dvc exp run --queue -S train.min_split=128
Queued experiment 'dbca012' for future execution.
$ dvc exp show --include-params=train.min_split --no-pager
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Experiment            ┃ Created      ┃ State  ┃ avg_prec ┃ roc_auc ┃ train.min_split ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ workspace             │ -            │ -      │  0.57553 │ 0.94652 │ 2               │
│ master                │ Aug 02, 2021 │ -      │  0.53252 │  0.9107 │ 2               │
│ ├── dbca012           │ 04:57 PM     │ Queued │        - │       - │ 128             │
│ ├── 8de9a6c [split16] │ 04:57 PM     │ Queued │        - │       - │ 16              │
│ ├── 5751540 [split32] │ 04:57 PM     │ Queued │        - │       - │ 32              │
│ └── e41d5b4 [split64] │ 04:57 PM     │ Queued │        - │       - │ 64              │
└───────────────────────┴──────────────┴────────┴──────────┴─────────┴─────────────────┘

> $ dvc exp remove dbca012
> $ dvc exp remove split16
> $ dvc exp show --include-params=train.min_split --no-pager
> ┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
> ┃ Experiment            ┃ Created      ┃ State  ┃ avg_prec ┃ roc_auc ┃ train.min_split ┃
> ┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
> │ workspace             │ -            │ -      │  0.57553 │ 0.94652 │ 2               │
> │ master                │ Aug 02, 2021 │ -      │  0.53252 │  0.9107 │ 2               │
> │ ├── 5751540 [split32] │ 04:57 PM     │ Queued │        - │       - │ 32              │
> │ └── e41d5b4 [split64] │ 04:57 PM     │ Queued │        - │       - │ 64              │
> └───────────────────────┴──────────────┴────────┴──────────┴─────────┴─────────────────┘

Reviewer: Again use less exps and use separate `table` code block.
````suggestion
$ dvc exp e41d5b4 split16

$ dvc exp show --include-params=train.min_split --no-pager
```

```table
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Experiment            ┃ Created      ┃ State  ┃ avg_prec ┃ roc_auc ┃ train.min_split ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ workspace             │ -            │ -      │  0.57553 │ 0.94652 │ 2               │
│ master                │ Aug 02, 2021 │ -      │  0.53252 │  0.9107 │ 2               │
│ └── 5751540 [split32] │ 04:57 PM     │ Queued │        - │       - │ 32              │
└───────────────────────┴──────────────┴────────┴──────────┴─────────┴─────────────────┘
```

```dvc
````

```

Nothing is listed after the last `dvc exp list` because they're all gone.

> The same applies to queued experiments but they won't have a name to give to
> `dvc exp remove` yet unless you specify one. Alternatively, you can use their ID
> instead (shown when queued and by `dvc exp show`). Let's queue a few experiments
> in different ways and then delete some of them:

Reviewer: ```suggestion
The same applies to queued experiments but these won't have a name to give to
`dvc exp remove` yet (unless you specified one). Alternatively, you can use their ID
(shown when queued, and by `dvc exp show`). Let's queue a few experiments
and then delete some of them:
```

```dvc
$ dvc exp run --queue -S train.min_split=64
Queued experiment 'e41d5b4' for future execution.
$ dvc exp run --queue -S train.min_split=32 --name split32
Queued experiment '5751540' for future execution.
$ dvc exp run --queue -S train.min_split=16 --name split16
Queued experiment '8de9a6c' for future execution.

> $ dvc exp remove e41d5b4 split16

Reviewer: ```suggestion
$ dvc exp remove e41d5b4 split16

```

```

Nothing is listed after the last `dvc exp list` because they're all gone.

> The same applies to queued experiments but these won't have a name to give to
> `dvc exp remove` yet (unless you specified one). Alternatively, you can use their ID
> (shown when queued, and by `dvc exp show`). Let's queue a few experiments
> and then delete some of them:

Reviewer: ```suggestion
The same applies to queued experiments but these won't have a name to give to
`dvc exp remove` yet (unless you specified one). Alternatively, you can use
their ID (shown when queued, and by `dvc exp show`). Let's queue a few
experiments and then delete some of them:
```
